packages/datatrak-web/src/views/AccountSettingsPage/PersonalDetailsSection.tsx

```diff
@@ -4,10 +4,168 @@
  */
 
 import React from 'react';
+import styled from 'styled-components';
+import { useForm, Controller, SubmitHandler } from 'react-hook-form';
 import { AccountSettingsSection } from './AccountSettingsSection';
+import { Button } from '../../components';
+import { successToast } from '../../utils';
+import { TextField } from '@tupaia/ui-components';
+import { useCurrentUser } from '../../api';
+import { useEditUser } from '../../api/mutations';
+import { UserAccountDetails } from '../../types';
+
+type PersonalDetailsFormFields = Pick<
+  UserAccountDetails,
+  'firstName' | 'lastName' | 'employer' | 'position' | 'mobileNumber'
+>;
+
+const ButtonWrapper = styled.div`
+  grid-column: -2;
+`;
+
+const StyledTextField = styled(TextField)`
+  margin: 0; // Use gap on parent to control spacing
+`;
+
+const PersonalDetailsForm = styled.form`
+  display: grid;
+  gap: 1.56rem 1.25rem;
+  max-width: 44.25rem;
+  width: 100%;
+  ${({ theme }) => theme.breakpoints.up('sm')} {
+    grid-template-columns: repeat(2, 1fr);
+  }
+
+  .MuiFormLabel-root {
+    color: ${props => props.theme.palette.text.primary};
+    font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
+  }
+`;
 
 export const PersonalDetailsSection = () => {
+  const { mutate: updateUser } = useEditUser(() => successToast('Personal details updated'));
+  const user = useCurrentUser();
+
+  const {
+    control,
+    formState: { isDirty, dirtyFields },
+    handleSubmit,
+    register,
+  } = useForm<PersonalDetailsFormFields>({
+    defaultValues: {
+      firstName: user.firstName ?? '',
+      lastName: user.lastName ?? '',
+      mobileNumber: user.mobileNumber ?? '',
+      employer: user.employer ?? '',
+      position: user.position ?? '',
+    },
+  });
+
+  function onSubmit(userDetails): SubmitHandler<PersonalDetailsFormFields> {
+    const updates = Object.keys(dirtyFields)
+      .filter(field => dirtyFields[field])
+      .reduce(
+        (updatedFields, field) => ({ ...updatedFields, [field]: userDetails[field] }),
+        {} as UserAccountDetails,
+      );
+
+    updateUser(updates);
+  }
+
   return (
-    <AccountSettingsSection title="Personal details" description="Edit your personal details" />
+    <AccountSettingsSection title="Personal details" description="Edit your personal details">
+      <PersonalDetailsForm onSubmit={handleSubmit(details => onSubmit(details))}>
+        <Controller
+          control={control}
+          name="firstName"
+          as={
+            <StyledTextField
+              label="First name"
+              placeholder="First name"
+              autoComplete="given-name"
+              inputProps={{ enterKeyHint: 'next' }}
```

Comment:
React complains about the way I specify the enter key hint:

```
React does not recognize the `enterKeyHint` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `enterkeyhint` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```

But inspection of the rendered `<input>` element shows that it works as expected. Is there a more paradigmatically correct way for me to do this? I would test on my tablet or phone if I knew how to deploy it over my local network 😅

Reply:
I think the problem is jus the casing as the html attribute is `enterkeyhint` all lowercase. I'm not familiar with the attribute but I would prefer to not include the attribute altogether as I'm not sure it's correct since hitting the enter key will submit the form, right?

Reply:
On desktop, yes, hitting <kbd>Enter</kbd> should submit the form.

This attribute is just for virtual keyboards (primarily phones and tablets, but also on-screen keyboards on desktop), and it just changes the **label** (not the behaviour) on the <kbd>Enter</kbd> so it looks like a <kbd>Next</kbd>/<kbd>Done</kbd>/<kbd>Send</kbd> key or whatever.

Personally I would argue that semantically labelling the key makes the UX (marginally) nicer, but if it continues to be more trouble that it’s worth, I’m happy to remove it entirely

Reply:
oh that's all good. I think not a bug priority but if simply changing it to lowercase fixes it then might as well keep it

Reply:
Based on https://github.com/necolas/react-native-web/pull/1707, it looks like the current implementation *works* correctly, and the warning is suppressed in newer versions of React.